src/pages/JoinEvent/JoinEvent.tsx

```diff
@@ -0,0 +1,216 @@
+import DashboardNavbar from "../../components/DashboardNavbar/DashboardNavbar";
+import { useAuth } from "../../context/AuthContext";
+
+const JoinEvent = () => {
+  const { eventId } = useParams();
+  const navigate = useNavigate();
+  const { currentUser } = useAuth();
+
+  const [name, setName] = useState("");
+  const [location, setLocation] = useState("");
+  const [timezone, setTimezone] = useState("(GMT-0700) Canada (Vancouver) Time");
+  const [skillLevel, setSkillLevel] = useState("Beginner");
+  const [discipline, setDiscipline] = useState("Developer");
+  const [goal, setGoal] = useState("");
+  const [strength, setStrength] = useState("");
+  const [weakness, setWeakness] = useState("");
+  const [eventData, setEventData] = useState(null);
+  const [isLoading, setIsLoading] = useState(true);
+
+  useEffect(() => {
+    const fetchEventData = async () => {
+      try {
+        const docRef = doc(db, "In_hackathonEvents", eventId);
+        const docSnap = await getDoc(docRef);
+        if (docSnap.exists()) {
+          setEventData(docSnap.data());
+        } else {
+          console.log("No such document!");
+          navigate("/hackathons");
+        }
+      } catch (error) {
+        console.error("Error fetching event data: ", error);
+        navigate("/hackathons");
+      } finally {
+        setIsLoading(false);
+      }
+    };
+
+    fetchEventData();
+  }, [eventId, navigate]);
+
+  const handleSubmit = async (event: FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+    if (!currentUser || !eventId) return;
+
+    const participantData = {
+      name,
+      location,
+      timezone,
+      skillLevel,
+      discipline,
+      goal,
+      strength,
+      weakness
+    };
+
+    try {
+      const eventRef = doc(db, "In_hackathonParticipantData", eventId);
+      const eventSnap = await getDoc(eventRef);
+
+      if (eventSnap.exists() && eventSnap.data().participants?.[currentUser.uid]) {
+        alert("You are already registered for this event.");
+        return;
+      }
+
+      await setDoc(eventRef, {
+        participants: {
+          [currentUser.uid]: participantData
+        }
+      }, { merge: true });
+
+      console.log("Participant data saved successfully");
+    } catch (error) {
+      console.error("Error saving participant data:", error);
+    }
+  };
+
+  if (!eventData) {
+    return <div>Loading...</div>;
+  }
+
+  return (
+    <div className="event-page">
+      <DashboardNavbar />
+      <div className="event-joinpage" style={{ height: "3rem", backgroundColor: "grey" }}>
+        <Link to="/hackathons" className="join-event__back-link">← Back</Link>
+      </div>
+      <div className="join-event">
+        <h1 className="join-event__title">Join {eventData.title}</h1>
+        <form className="join-event__form" onSubmit={handleSubmit}>
+        <div className="join-event__form-group">
+          <label htmlFor="name" className="join-event__label">Name*</label>
```

Comment:
The labels with the * should have 'required' added to their input elements, but I realize you probably omitted it for testing.